### PR TITLE
[msg] Fix counter init.

### PR DESCRIPTION
**Pull request by turon**
#### Problem
Specification updated some details on message initialization to fix rare but possible off-by-one errors.

#### Change overview
Implement fixes to message initialization mandated by spec.

#### Testing
local and CI

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: src/transport/MessageCounter.h [hunk 1]
```diff
@@ -88,44 +88,44 @@ class GlobalEncryptedMessageCounter : public MessageCounter
...
-    static constexpr uint32_t kInitialValue                 = 1;         ///< Used for initializing peer counter
+    static constexpr uint32_t kInitialSyncValue             = 0;         ///< Used for initializing peer counter
     static constexpr uint32_t kMessageCounterRandomInitMask = 0x0FFFFFF; ///< 28-bit mask
 
     /**
      * Initialize a local message counter with random value between [0, 2^28-1]. This increases the difficulty of traffic analysis
      * attacks by making it harder to determine how long a particular session has been open.
      */
-    LocalSessionMessageCounter() { value = Crypto::GetRandU32() & kMessageCounterRandomInitMask; }
+    LocalSessionMessageCounter() { mValue = (Crypto::GetRandU32() + 1) & kMessageCounterRandomInitMask; }
```
Comment (andy31415) [inline src/transport/MessageCounter.h:117]: Don't `GetRandU32()` and `GetRandU32() + 1` have  an identical distribution given that we mask them (i.e. they will effectively wrap over)?

I do not understand this change and it makes the code more complex. If this is actually intended (I am really unclear how that could be though ... chances of a 0 should be the same as for a 0xFFFFFF) please add a comment.

Comment (andy31415) [reply src/transport/MessageCounter.h:117]: I tried comparing something like this (much smaller bin size to actually see results):

```py
import matplotlib.pyplot as plt
import random
%matplotlib inline

plt.hist([random.randint(0, 0xFFFFFFFF) & 0xF for x in range(200000)], bins=range(0x0F + 2))
plt.hist([(random.randint(0, 0xFFFFFFFF) + 1) & 0xF for x in range(200000)], bins=range(0x0F + 2))
```

they are the same (except slight differences due to random ... for sure no difference around 0 or 0x0F

**Review by andy31415 — CHANGES_REQUESTED** [review]
Request changes: please prove that `GetRandU32() + 1` actually does something useful (show me python code that shows relevance) or remove that change.

File: src/transport/MessageCounter.h (hunk 1 above)
Comment (turon) [reply src/transport/MessageCounter.h:117]: My mistake, need to increment after mask, will fix.

**Review by msandstedt — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by emargolis — APPROVED** [review]

**Pull request merged**